evaluator/evaluate.go

```diff
@@ -211,23 +212,28 @@ type namedPackage struct {
 }
 
 // Find relevant package data based on vmaas results
-func getPackages(tx *gorm.DB, data vmaas.UpdatesV2Response) (map[utils.Nevra]models.Package, error) {
-	pkgNevras := make([]string, 0, len(data.UpdateList))
+func loadPackages(tx *gorm.DB, data vmaas.UpdatesV2Response) (map[utils.Nevra]models.Package, error) {
+	names := make([]string, 0, len(data.UpdateList))
+	evras := make([]string, 0, len(data.UpdateList))
+	defer utils.ObserveSecondsSince(time.Now(), evaluationPartDuration.WithLabelValues("packages-load"))
 	for nevra := range data.UpdateList {
 		// Parse and reformat nevras to avoid issues with 0 epoch
 		parsed, err := utils.ParseNevra(nevra)
 		if err != nil {
 			utils.Log("err", err.Error(), "nevra", nevra).Warn("Unable to parse nevra")
 			continue
 		}
-		pkgNevras = append(pkgNevras, parsed.String())
+		names = append(names, parsed.Name)
+		evras = append(evras, parsed.EVRAString())
 	}
-
+	// Might return more data than we need (one EVRA being applicable to more packages
+	// But it was only way to get somewhat fast query plan which only uses index scans
 	var packages []namedPackage
 	err := tx.Table("package").
 		Select("concat(pn.name, '-', package.evra) nevra, package.*").
 		Joins("join package_name pn on package.name_id = pn.id").
-		Where("concat(pn.name, '-', package.evra) in (?)", pkgNevras).Find(&packages).Error
+		Where("pn.name in (?)", names).
```

Comment:
Maybe it would be better not to extract package name into the separate table but rather keep it as string column in the package table and then create multi-column index on package.name and package.evra, what do you think?

Reply:
We still have same problem, we need to search for specific name-evra pairs, and we can't do that with current library without performing a full table scan.